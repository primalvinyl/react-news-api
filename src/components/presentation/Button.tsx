import React from 'react';
import styles from './Button.module.scss';

const Button = React.forwardRef<HTMLButtonElement, ButtonType>(
    (props, ref): React.ReactElement => {

    const {
        id,
        className,
        name,
        value,
        type,
        disabled,
        isSubmitting,
        color,
        url,
        handleClick
    } = props;

    const onClick = (event: React.MouseEvent<HTMLButtonElement>) => {
        if (handleClick) handleClick(event);
        if (url) window.open(url, 'article', 'noopener,noreferrer');
    };

    return (
        <div className={styles.root}>
            <button
                className={className}
                id={id}
                name={name}
                type={type}
                disabled={isSubmitting || disabled}
                ref={ref}
                style={{ background: color ? color : styles.gray800 }}
                onClick={onClick}
            >
                {value}
            </button>
        </div>
    );
});

type ButtonType = {
    readonly id: string;
    readonly className?: string;
    readonly name?: string;
    readonly value?: string;
    readonly type?: 'submit' | 'button' | 'reset';
    readonly disabled?: boolean;
    readonly isSubmitting?: boolean;
    readonly color?: string;
    readonly url?: string;
    readonly handleClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
};

Button.defaultProps = {
    className: undefined,
    name: undefined,
    value: '',
    type: 'submit',
    disabled: false,
    isSubmitting: false,
    color: '',
    url: undefined,
    handleClick: (): void => {},
}

export default Button;
