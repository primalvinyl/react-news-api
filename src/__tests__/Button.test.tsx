import * as React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import Button from '../components/presentation/Button';

describe('Button', () => {
    it('renders', () => {
        const myWrapper = shallow(<Button id="test" />);
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(<Button id="test" />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});