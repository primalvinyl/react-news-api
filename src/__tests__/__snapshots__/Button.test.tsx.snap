// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div
  className="root"
>
  <button
    disabled={false}
    id="test"
    onClick={[Function]}
    style={
      Object {
        "background": "gray800",
      }
    }
    type="submit"
  >
    
  </button>
</div>
`;
